Sources/SwiftDocC/Infrastructure/Diagnostics/DiagnosticConsoleWriter.swift

```diff
@@ -165,15 +189,269 @@ struct IDEDiagnosticConsoleFormatter: DiagnosticConsoleFormatter {
+
+            for (columnNumber, columnSuggestions) in suggestionsPerColumn.sorted(by: { $0.0 > $1.0 }) {
+                var prefix = suggestionLinePrefix
+
+                for column in 0...columnNumber - 1 {
+                    if columnsWithSuggestions.contains(column) {
+                        prefix.append("│")
+                    } else {
+                        prefix.append(" ")
+                    }
+                }
+
+                for (index, suggestion) in columnSuggestions.enumerated() {
+                    // Highlight suggestion and make sure it's displayed on a single line.
+                    let singleLineSuggestion = suggestion.split(separator: "\n", omittingEmptySubsequences: true).joined(separator: "")
+                    let highlightedSuggestion = highlightSuggestion("suggestion: \(singleLineSuggestion)")
+
+                    if index == columnSuggestions.count - 1 {
+                        result.append("\n\(prefix)╰─\(highlightedSuggestion)")
+                    } else {
+                        result.append("\n\(prefix)├─\(highlightedSuggestion)")
+                    }
+                }
+            }
+        }
+        
+        return result
+    }
+    
+    private func highlightSuggestion(
+        _ suggestion: String
+    ) -> String {
+        guard highlight
+        else { return suggestion }
+        
+        let suggestionAnsiAnnotation = ANSIAnnotation.sourceSuggestionHighlight
+        return suggestionAnsiAnnotation.applied(to: suggestion)
+    }
+
+    private func highlightSource(
+        sourceLine: String,
+        lineNumber: Int,
+        range: SourceRange
+    ) -> String {
+        guard highlight,
+              lineNumber >= range.lowerBound.line && lineNumber <= range.upperBound.line,
+              !sourceLine.isEmpty
+        else { return sourceLine }
+        
+        var startColumn: Int
+        if lineNumber == range.lowerBound.line {
+            startColumn = range.lowerBound.column
+        } else {
+            startColumn = 1
+        }
+        
+        var endColumn: Int
+        if lineNumber == range.upperBound.line {
+            endColumn = range.upperBound.column
+        } else {
+            endColumn = sourceLine.count + 1
+        }
+        
+        let columnRange = startColumn..<endColumn
+        let startIndex = sourceLine.index(sourceLine.startIndex, offsetBy: columnRange.lowerBound - 1)
+        let endIndex = sourceLine.index(startIndex, offsetBy: columnRange.count)
+        
+        let ansiAnnotation = ANSIAnnotation.sourceHighlight
+
+        var result = ""
+        result += sourceLine[sourceLine.startIndex..<startIndex]
+        result += ansiAnnotation.applied(to: String(sourceLine[startIndex..<endIndex]))
+        result += sourceLine[endIndex..<sourceLine.endIndex]
+        
+        return result
+    }
+
+    private func readSourceLines(_ url: URL) -> [String] {
```

Comment:
I don't think this would be able to read the relevant lines from a symbol's in-source documentation comment if the original source file wasn't there. When DocC is used as part of a build workflow the original source files would be present at their original locations but it's also possible to build documentation with a symbol graph file without the original source being there (and the symbol graph file contains a copy of what each symbol's documentation comment was). 

I don't think this needs to be fixed in this PR but it would be good with a todo comment that mentions adding support for getting the source lines from the symbol graph files.

Reply:
Great point, I did not consider this scenario. I added a todo comment to add support for this at a later time!